### PR TITLE
Add extra files in bundle (backend)

### PR DESCRIPTION
**Pull request by mmarchetti**
### Description

This PR adds an optional `files` parameter to the deploy endpoint JSON. This is an array of  file paths relative to the notebook directory. Paths are resolved and checked to ensure that they are in the notebook directory.

Connected to #https://github.com/rstudio/connect/issues/15698

### Testing Notes / Validation Steps
Requires the frontend piece to fully test. You can simulate this by modifying `rsconnect.js`:

```diff
diff --git rsconnect_jupyter/static/rsconnect.js rsconnect_jupyter/static/rsconnect.js
index 6ac263a..df8a346 100644
--- rsconnect_jupyter/static/rsconnect.js
+++ rsconnect_jupyter/static/rsconnect.js
@@ -426,7 +426,8 @@ define([
                         include_files: includeFiles,
                         include_subdirs: includeSubdirs,
                         disable_tls_check: entry.disableTLSCheck,
-                        cadata: self.getCAData(entry.server)
+                        cadata: self.getCAData(entry.server),
+                        files: ['path/to/file']
                     };

                     var xhr = Utils.ajax({
```

### CODE REVIEW

**Comment by mmarchetti** [comment]
Yes, effectively. This PR doesn't remove the frontend checkboxes, so I plan to hold off merging until the frontend is ready (otherwise master will be in a broken state). Or I could remove the boxes in this PR if you prefer.

**Comment by scottmmjackson** [comment]
It might make more sense if we leave this open and I stage my PR as a PR to this branch.

**Comment by mmarchetti** [comment]
I agree

**Comment by scottmmjackson** [comment]
@mmarchetti whenever you're back, I think this is good for review.

However, I did notice I had a problem deploying my `Untitled.ipynb` due to the fact that it wouldn't restore `matplotlib`. Not sure if the file inclusion logic messed with that.

**Comment by scottmmjackson** [comment]
This is not ready for merge, there are some bugs we need to fix. Stay tuned...

**Comment by scottmmjackson** [comment]
Bugs should be fixed.

File: rsconnect_jupyter/static/connect.js [hunk 1]
```diff
@@ -21,6 +22,13 @@ define([
     Canceled: 'canceled'
   };
 
+  var ContentsManager = new Contents.Contents({ base_url: Jupyter.notebook.base_url });
+  var lastSlashInNotebookPath = Jupyter.notebook.notebook_path.lastIndexOf('/');
+  var notebookDirectory = '';
+  if (lastSlashInNotebookPath !== -1) {
+    notebookDirectory = Jupyter.notebook.notebook_path.slice(0, Jupyter.notebook.notebook_path.lastIndexOf('/'));
```
Comment (mmarchetti) [inline rsconnect_jupyter/static/connect.js]: nit: this could use `lastSlashInNotebookPath` instead of calling `lastIndexOf` again.

**Comment by scottmmjackson** [comment]
The CI failure is stochastic. Merging over it.

**Pull request merged**